src/java.base/share/classes/jdk/internal/module/ModuleReferenceImpl.java

```diff
@@ -139,13 +149,19 @@ public ModuleResolution moduleResolution() {
      * @throws java.io.UncheckedIOException if an I/O error occurs
      */
     public byte[] computeHash(String algorithm) {
-        byte[] result = cachedHash;
-        if (result != null)
-            return result;
-        if (hasher == null)
+        CachedHash ch = cachedHash;
+        if (ch != null) {
+            if (ch.algorithm.equals(algorithm)) {
```

Comment:
The nested if looks a bit strange here, you can change to:
```
if (ch != null & ch.algorithm.equals(algorithm))
    return ch.hash;
```

Reply:
Already done.